iothub/device/tests/Authentication/EdgeDeviceOnBehalfOfTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text;
+using System.Threading.Tasks;
+using System;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client.Authentication;
+
+namespace Microsoft.Azure.Devices.Client.Tests.OnBehalfOf
+{
+    public class EdgeDeviceOnBehalfOfTests
+    {
+        private static string _testKey => Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.Empty.ToString("N")));
+
+        [TestMethod]
+        public async Task ConnectDeviceOnBehalfOf()
+        {
+            string leafDeviceId = "test-leaf-device";
+
+            DeviceConnectionSettings edgeDevice = new DeviceConnectionSettings("e4k-hub.azure-devices.net", "test-edge-device")
+            {
+                SharedAccessKey = _testKey
+            };
+
+            var edgeHubCs = await EdgeHubConnectionSettingsBuilder.Build(edgeDevice);
+
+            IAuthenticationMethod leafAuth = new ClientAuthenticationForEdgeHubOnBehalfOf(
+                edgeHubCs.SharedAccessKey!,
+                edgeHubCs.DeviceId!,
+                leafDeviceId,
+                null,
+                TimeSpan.FromMinutes(10),
+                5);
+
+            IotHubDeviceClient leafClient = new(edgeHubCs.HostName, leafAuth,
+                new IotHubClientOptions(
+                    new IotHubClientAmqpSettings
+                    {
+                        ConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                        {
+                            UsePooling = true,
+                            MaxPoolSize = 10
+                        }
+                    }));
+
+            await leafClient.OpenAsync();
+            long tick = Environment.TickCount;
+            await leafClient.UpdateReportedPropertiesAsync(new ReportedProperties { ["tick"] = tick });
+            var twin = await leafClient.GetTwinPropertiesAsync();
+            twin.Should().NotBeNull();
+            twin.Reported.Should().NotBeNull();
+            twin.Reported["tick"].Should().NotBeNull();
+            tick.Should().Be((long)twin.Reported["tick"]);
+            await leafClient.CloseAsync();
+        }
+
+        [TestMethod]
+        public async Task ConnectLeafModuleOnBehalfOf()
+        {
+            string leafDeviceId = "test-leaf-device";
+            string edgeModuleId = "test-leaf-device-module";
+
+            DeviceConnectionSettings edgeDevice = new DeviceConnectionSettings("e4k-hub.azure-devices.net", "test-edge-device")
+            {
+                SharedAccessKey = _testKey
+            };
+
+            var edgeHubCs = await EdgeHubConnectionSettingsBuilder.Build(edgeDevice);
```

Comment:
Where is EdgeHubConnectionSettingsBuilder? bringing this feature to V2 is very compelling (to enable provision any module, and initialize edgeHub), but I would leave it for a different PR.

@abhipsaMisra  ?

Reply:
btw.. the tests can be initialized with a _known_  $edgehub key